stl/inc/mdspan

```diff
@@ -11,11 +11,8 @@
 #if !_HAS_CXX23 || !defined(__cpp_lib_concepts) // TRANSITION, GH-395
 _EMIT_STL_WARNING(STL4038, "The contents of <mdspan> are available only with C++23 or later.");
 #else // ^^^ not supported / supported language mode vvv
-#include <algorithm>
 #include <array>
 #include <span>
-#include <tuple>
-#include <type_traits>

```

Comment:
Is there a reason to remove `type_traits`, given it's still used in the file?

Reply:
`<type_traits>` is already included by both `<array>` and `<span>`, but if it is necessary then I can restore it.

Reply:
> `<type_traits>` is already included by both `<array>` and `<span>`
Better to never rely on implicit inclusions that may change in the future.

Reply:
OK, I'm going to restore this `#include` in next PR (to avoid restarting whole CI).

Reply:
> > `<type_traits>` is already included by both `<array>` and `<span>`
> 
> Better to never rely on implicit inclusions that may change in the future.

This doesn't seem an issue since the MSVC STL is maintained as a whole, although it makes sense that STL shouldn't rely on the implicit inclusions in UCRT or VCRuntime. If the inclusion relation is changed in future, we can change `<mdspan>` in the same PR.